helpers/postgres.rb

```diff
@@ -56,8 +56,20 @@ def postgres_post(name)
     end
   end
 
-  def postgres_list
+  def postgres_list(tags_param: nil)
     dataset = dataset_authorize(@project.postgres_resources_dataset.eager(:timeline, representative_server: [:strand, vm: :vm_storage_volumes]), "Postgres:view").eager(:semaphores, :location, strand: :children)
+
+    if tags_param
+      tags_param = tags_param.split(",").map { |tag| tag.split(":", 2).map(&:strip) }
+      tags_param = tags_param.map { |key, value| {key:, value:} }
+      tags_param.each do |tag|
+        if tag.length != 2 || tag[:key].nil? || tag[:value].nil?
+          fail Validation::ValidationFailed.new({tags: "Invalid tag format. Expected format: key:value"})
+        end
+      end
+      dataset = dataset.where(Sequel.pg_jsonb_op(:tags).contains(tags_param))
```

Comment:
for my understanding: these tags could instead be setup with ACL so result would filter out the rest, tho even in that case it'd be useful to have this as an extra layer in case ACL is misconfigured (or if api user wants to filter tags for a subset of their pg resources ofc)

Reply:
Yes and no. We could have tags via ACL but that would not be these tags, it would be `lib/AccessControlModelTag`. Since that tag implementation is not very commonly used and we already have a separate `tags` column here in PostgresResource, we opted to use this one for the time being. In future, if we want to move, we could simply migrate these tags to AccessControlModelTags and start using that as a single source of truth.